File: 1013-fibonacci-number/fibonacci-number.c
int fib(int n){
    if(n == 0|| n == 1 ) return n;
    int a = 0,b= 1;
    int sum;
    for(int i=2;i< n + 1;i++ ){
        sum = a + b;
        a = b;
        b = sum;
    }
    return sum;
}

File: 1013-fibonacci-number/fibonacci-number.py
class Solution:
    def fib(self, n: int) -> int:
        dp = []
        if n <= 1:
            return n
        for i in range(n+1):
            if i <= 1:
                dp.append(i)
            else:
                dp.append(dp[i-1] + dp[i-2])
        return dp[n]